source/cmake_modules/ThirdParty.cmake

```diff
@@ -0,0 +1,64 @@
+# Third party
+
+# Third party warnings have to be silent, we skip warnings
+set(TMP_FLAG ${CMAKE_CXX_FLAGS})
+if(CMAKE_COMPILER_IS_GNUCXX)
+    set(CMAKE_CXX_FLAGS "${CMAKE_GNUCXX_COMMON_FLAGS} -w")
+endif(CMAKE_COMPILER_IS_GNUCXX)
+
+if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CLANG_COMMON_FLAGS} -Wno-everything")
+endif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+
+#
+# SimpleAmqpClient
+# librabbimq-c
+#
+include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/third_party/SimpleAmqpClient/src/")
+set(BUILD_STATIC_LIBS ON CACHE BOOL "Build Rabbitmqc as static library")
+set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build SimpleAmqpClient as static library")
+set(BUILD_API_DOCS OFF CACHE BOOL "don't build doc of librabbimq-c")
+set(BUILD_EXAMPLES OFF CACHE BOOL "don't build example of librabbimq-c")
+set(BUILD_TOOLS OFF CACHE BOOL "don't build tool of librabbimq-c")
+set(BUILD_TESTS OFF CACHE BOOL "don't build tests of librabbimq-c")
+set(CMAKE_INSTALL_PREFIX "")
+set(Rabbitmqc_INCLUDE_DIRS "${CMAKE_SOURCE_DIR}/third_party/librabbitmq-c/librabbitmq/")
+add_subdirectory(third_party/librabbitmq-c)
+add_subdirectory(third_party/SimpleAmqpClient)
+
+#
+# Include third_party code in system mode, to avoid warnings
+#
+include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/third_party/")
+
+#
+# prometheus-cpp
+#
+# We want to deactivate prometheus lib if CMake version < 2.8.12.2
+if(("${CMAKE_VERSION}" VERSION_GREATER 2.8.12.2) OR ("${CMAKE_VERSION}" VERSION_EQUAL 2.8.12.2))
```

Comment:
Ugly... But looks like it's the way :-(
Maybe NOT VERSION_LESS, or VERSION_GREATER 2.8.12.1 ?

Reply:
It's a good point, VERSION_GREATER 2.8.12.1 is more compact.